www/runtest.php

```diff
@@ -2,920 +2,925 @@
-      if (isset($profiles) && is_array($profiles) && isset($profiles[$_REQUEST['webvital_profile']])) {
-        foreach($profiles[$_REQUEST['webvital_profile']] as $key => $value) {
-          if ($key !== 'label' && $key !== 'description') {
-            $_REQUEST[$key] = $value;
-            $_GET[$key] = $value;
-          }
-        }
-      }
+}
+$wvprofile_file = __DIR__ . '/settings/profiles_webvitals.ini';
+if (file_exists(__DIR__ . '/settings/common/profiles_webvitals.ini'))
+    $wvprofile_file = __DIR__ . '/settings/common/profiles_webvitals.ini';
+if (file_exists(__DIR__ . '/settings/server/profiles_webvitals.ini'))
+    $wvprofile_file = __DIR__ . '/settings/server/profiles_webvitals.ini';
+if (isset($_REQUEST['webvital_profile']) && is_file($wvprofile_file)) {
+    $profiles = parse_ini_file($wvprofile_file, true);
+    if (isset($profiles) && is_array($profiles) && isset($profiles[$_REQUEST['webvital_profile']])) {
+        foreach ($profiles[$_REQUEST['webvital_profile']] as $key => $value) {
+            if ($key !== 'label' && $key !== 'description') {
+                $_REQUEST[$key] = $value;
+                $_GET[$key] = $value;
+            }
+        }
     }
-    require_once('common.inc');
+}
+require_once('common.inc');
 
 use WebPageTest\Util;
 use WebPageTest\Template;
 use WebPageTest\RateLimiter;
 
-    require_once('./ec2/ec2.inc.php');
-    require_once(__DIR__ . '/include/CrUX.php');
-
-    $experimentURL = Util::getSetting('experimentURL', null);
-    $ui_priority = !is_null($request_context->getUser()) ? $request_context->getUser()->getUserPriority() : 0;
-
-    set_time_limit(300);
-
-    $redirect_cache = array();
-    $error = NULL;
-    $xml = false;
-    $usingAPI = false;
-    $usingApi2 = false;
-    $forceValidate = false;
-    $runcount = 0;
-    $apiKey = null;
-    $is_mobile = false;
-    $isPaid = !is_null($request_context->getUser()) && $request_context->getUser()->isPaid();
-    $includePaid = $isPaid || $admin;
-    // load the secret key (if there is one)
-    $server_secret = Util::getServerSecret();
-    $api_keys = null;
-    if (isset($_REQUEST['k']) && strlen($_REQUEST['k'])) {
-      $keys_file = __DIR__ . '/settings/keys.ini';
-      if (file_exists(__DIR__ . '/settings/common/keys.ini'))
+require_once('./ec2/ec2.inc.php');
+require_once(__DIR__ . '/include/CrUX.php');
+
+$experimentURL = Util::getSetting('experimentURL', null);
+$ui_priority = !is_null($request_context->getUser()) ? $request_context->getUser()->getUserPriority() : 0;
+
+set_time_limit(300);
+
+$redirect_cache = array();
+$error = NULL;
+$xml = false;
+$usingAPI = false;
+$usingApi2 = false;
+$forceValidate = false;
+$runcount = 0;
+$apiKey = null;
+$is_mobile = false;
+$isPaid = !is_null($request_context->getUser()) && $request_context->getUser()->isPaid();
+$hasRunsAvailable = !is_null($request_context->getUser()) && $request_context->getUser()->hasEnoughRemainingRuns();
```

Comment:
Unless we pass in the total runs being run, this will test if the person has more than 1 run.